src/test/javascript/integration/item_length_large_spec.js

```diff
@@ -4,45 +4,60 @@ const {
     getHubItem,
     getProp,
     hubClientDelete,
+    hubClientGet,
     hubClientPost,
     randomChannelName,
 } = require('../lib/helpers');
 const {
+    getHubUrlBase,
     getChannelUrl,
 } = require('../lib/config');
 
 const channelUrl = getChannelUrl();
+const channelName = randomChannelName();
+const channelEndpoint = `${channelUrl}/${channelName}`;
+const itemHeaders = {'Content-Type': 'text/plain'};
+const headers = { 'Content-Type': 'application/json' };
+const testContext = {
+    itemSize: 0,
+    itemContent: '',
+    itemURL: null,
+    createdChannel: false,
+};
+
 /**
  * POST a large item, GET it, and verify the "X-Item-Length"
  * header is present with the correct value
  */
 
 describe(__filename, function () {
-    const channelName = randomChannelName();
-    const channelEndpoint = channelUrl + '/' + channelName;
-    const itemHeaders = {'Content-Type': 'text/plain'};
-    const itemSize = 41 * 1024 * 1024;
-    const itemContent = Array(itemSize).join('a');
-    let itemURL;
-    let createdChannel = false;
-
     beforeAll(async () => {
         const channel = await createChannel(channelName, null, 'large inserts');
         if (getProp('statusCode', channel) === 201) {
-            createdChannel = true;
+            testContext.createdChannel = true;
             console.log(`created channel for ${__filename}`);
         }
+        const propertiesUrl = `${getHubUrlBase()}/internal/properties`;
+        const internals = await hubClientGet(propertiesUrl, headers);
+        const properties = fromObjectPath(['body', 'properties'], internals) || {};
+        const maxPayloadSizeMB = parseInt(properties['app.maxPayloadSizeMB'] || '43');
+        testContext.itemSize = maxPayloadSizeMB - 1;
+        testContext.itemContent = Array(testContext.itemSize).join('a');
     });
 
     it('posts a large item', async () => {
+        const { createdChannel } = testContext;
         if (!createdChannel) return fail('channel not created in before block');
+        const { itemContent } = testContext;
         const response = await hubClientPost(channelEndpoint, itemHeaders, itemContent);
         const body = getProp('body', response);
-        itemURL = fromObjectPath(['_links', 'self', 'href'], body);
+        const itemURL = fromObjectPath(['_links', 'self', 'href'], body);
         expect(itemURL).toBeDefined();
+        testContext.itemURL = itemURL;
```

Comment:
These sequential tests kill me; we have to go and verify that it's defined in the next one, too. :sigh: